### PR TITLE
Feature/Core - Log checker message when collecting Issues

### PR DESCRIPTION
**Pull request by trgiangdo**
Used in https://github.com/Avaiga/taipy-core/pull/492 to run only the `_MigrationChecker` on production.

### CODE REVIEW

File: src/taipy/config/config.py [hunk 1]
```diff
@@ -175,10 +176,24 @@ def check(cls) -> IssueCollector:
...
             Collector containing the info, warning and error issues.
         """
-        cls._collector = _Checker._check(cls._applied_config)
-        cls.__log_message(cls)
+        cls._collector = _Checker._check_all(cls._applied_config)
+        cls.__log_message(cls._collector)
         return cls._collector
 
+    @classmethod
+    def _check_one(cls, checker: Type[_ConfigChecker]) -> IssueCollector:
```
Comment (trgiangdo) [inline src/taipy/config/config.py]: The naming is not perfect yet. Let me know what you think

Comment (jrobinAV) [reply src/taipy/config/config.py]: We are exposing an API just for a special case, in order to run one checker that is defined in a 'user' repository (taipy-core in our case). 

I believe we should not do that.

**Review by jrobinAV — COMMENTED** [review]
I would not change the checker APIs just for a special case in particular because the case consists in checking a core section defined in the Core repository and code.

I believe we should not merge the PR.

**Review by jrobinAV — CHANGES_REQUESTED** [review]
I would not change the checker APIs just for a special case in particular because the case consists in checking a core section defined in the Core repository and code.

I believe we should not merge the PR.

**Comment by trgiangdo** [comment]
In the latest commit, the `Config._check_one()` is removed.

To handle a separate checker on Core, the `Config.__log_message()` is removed, and we log the message when an Issue is added to the Collector.

**Review by jrobinAV — COMMENTED** [review]
I don't really like mixing the responsibilities of checking errors and logging them. I prefer to let the responsibility to the client for logging the messages.
For example, we can imagine a client not logging all issues. It may want to group them by type (or severity or whatever), and only log one issue per group and display the size of the group. This is pretty common, especially when there are a lot of issues.

Plus, I don't see any added value for the user.
The only difference I can see is really minor to me. It s just we don't need to log the migrationConfig issues in Core. Which I believe it s the right place.

I would personally just drop the PR. Sorry. What do you think? Do I miss a reason for such a change?

**Comment by trgiangdo** [comment]
You're right. I'll close the PR and move the log method to core for the MigrationConfigChecker.

Thank you for the review

**Pull request closed without merging**